### PR TITLE
Casper refactors encore

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by marcin-rzeznicki — APPROVED** [review]

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
@will I tried to update the integration tests, but `test_slash_invalid_block_seq` still fails on me. Could you have a look? I'm pretty sure the failure is due to one of the last two commits...

see https://drone.rchain-dev.tk/rchain/rchain/5956/7

**Comment by will** [comment]
Oh so the problem is that blockchian shit is really dumb and has nothing good about it at all, hth

**Comment by ArturGajowy** [comment]
Sorry for the wrong mention @will :)

@zsluedem I tried to update the integration tests, but test_slash_invalid_block_seq still fails on me. Could you have a look? I'm pretty sure the failure is due to one of the last two commits...

see https://drone.rchain-dev.tk/rchain/rchain/5956/7

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
I've removed the last commit (which I assume breaks the tests) to do a `bors try`.

Will open a separate PR for "Remove the unused `Header.deploysHash`" later

**Comment by ArturGajowy** [comment]
@tymm @JosDenmark @odeac ping :)

File: casper/src/main/scala/coop/rchain/casper/BlockCreator.scala [hunk 1]
```diff
@@ -213,45 +211,39 @@ object BlockCreator {
...
-                  shardId,
-                  version
-                )
-              }
+    } yield block)
+    //TODO both the log message and block status seems misleading - the error could have happened anywhere,
+    // e.g. during `replayIntoMergeBlock`
+      .onError {
+        case ex =>
+          Log[F].error(s"Critical error encountered while processing deploys: ${ex.getMessage}")
```
Comment (itegulov) [inline casper/src/main/scala/coop/rchain/casper/BlockCreator.scala]:
```suggestion
          Log[F].error(s"Critical error encountered while processing deploys", ex)
```

**Review by itegulov — APPROVED** [review]

**Comment by zsluedem** [comment]
@ArturGajowy Sorry. I didn't pay much attention on the github before. I saw you resolve the problem by removing commits. Did you change the protocol(protobuf) in the last commit?

**Comment by zsluedem** [comment]
I know where the problem is https://github.com/rchain/rchain/commit/5c8efc577ba5c539bbb109b6f5ce14a163cfa509#diff-0915fb7518162e911f6aff8931db4054. I can create a patch for you.

**Comment by zsluedem** [comment]
```
message HeaderProto {
  repeated bytes parentsHashList = 1; //list of parent block hashes
  int64 timestamp = 5;
  int64 version = 6;
  bytes extraBytes = 8;
}
```
Maybe can do the trick

File: casper/src/main/scala/coop/rchain/casper/BlockCreator.scala (hunk 1 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/BlockCreator.scala]: Include error cause stack trace in `processDeploysAndCreateBlock`

**Pull request merged**